### PR TITLE
ECB status and control messages

### PR DESCRIPTION
**Pull request by EShamaev**
First draft of messages for electronic circuit breaker systems.

Examples are:
http://verticalpower.com/index.php/products/vp-x
http://www.mglavionics.co.za/ecb.html
http://global.wago.com/en/products/new-items/overview/basic-page-50.jsp

### CODE REVIEW

**Comment by EShamaev** [comment]
@pavel-kirienko Please help with determing the maximum number of messages that can be contained in 1094.ArrayECBChannels.uavcan

File: uavcan/equipment/power/1093.ECB.uavcan [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Typical publishing rate should be around 1 Hz.
+#
+
+#
+# ECB technical charateristics
+#
+uint4 number_of_input_channels  # Total number of power supplies
+uint4 number_of_output_channels # Total number of switched power circuits
+uint8 nominal_voltage           # Nominal voltage of ECB (for example 14 or 28 VDC)
+uint8 maximum_total_current     # Maximum total current that ECB is capable withstand
```
Comment (pavel-kirienko) [inline uavcan/equipment/power/1093.ECB.uavcan]: I propose to make voltage and current of type `float16` for reasons of consistency. The added overhead is negligible.

File: uavcan/equipment/power/1093.ECB.uavcan [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+uint4 STATUS_FLAG_OVERHEAT     = 4 # Temperature of ECB is above limit
+uint4 status_flags
+
+#
+# ECB identification.
+# Model name is a human-readable string that normally should include the vendor name, model name and
+# type of this ECB. This field should be assumed case-insensitive. Example: "Zubax ECB-32 v1.2".
+#
+uint8 device_id          # Identifies the ECB within this vehicle
+uint8[<32] model_name    # ECB model name
```
Comment (pavel-kirienko) [inline uavcan/equipment/power/1093.ECB.uavcan]: Are you sure we need the model name here? Isn't it redundant considering that we have `GetNodeInfo.name`?

File: uavcan/equipment/power/1095.ECBChannelCommand.uavcan [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#
```
Comment (pavel-kirienko) [inline uavcan/equipment/power/1095.ECBChannelCommand.uavcan]: Make this a service.

File: uavcan/equipment/power/ECBChannel.uavcan [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+uint4 STATUS_FLAG_WIG_WAG       = 2 # Channel is ON and pulsing (wig wag).
+uint4 STATUS_FLAG_PULLED_OFF    = 4 # Channel is disabled to be turned on
+uint4 status_flags
+
+#
+# Setup of channel
+#
+float16 nominal_current       # [Ampere] Nominal current that this channel is set for
+float16 short_circuit_current # [Ampere] Current that will lead to immidiate tripping
+float16 IIt                   # [Ampere^2 * Second] Thermal protection charateristic
```
Comment (pavel-kirienko) [inline uavcan/equipment/power/ECBChannel.uavcan]: `iit`

File: uavcan/equipment/power/ECBInputChannel.uavcan [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+#
+
+#
+# Current status of channel
+#
+float16 voltage # Output voltage on channel
+float16 current # Current through channel
+
+uint2 ERROR_FLAG_OVERCURRENT    = 1 # Channel current is above maximum
+uint2 error_flags
```
Comment (pavel-kirienko) [inline uavcan/equipment/power/ECBInputChannel.uavcan]: Please make the next field byte-aligned. Perhaps throw in more bits to `error_flags`?

**Comment by pavel-kirienko** [comment]
Closing as stale.

**Pull request closed without merging**